python/cudf/cudf/_lib/csv.pyx

```diff
@@ -207,10 +188,34 @@ cpdef read_csv(
     if prefix is not None:
         args.prefix = prefix.encode()
 
-    cdef unique_ptr[cpp_csv.reader] reader
-    with nogil:
-        reader = unique_ptr[cpp_csv.reader](new cpp_csv.reader(args))
+    # Create reader from source
+    cdef unique_ptr[csv_reader] reader
+    cdef const unsigned char[::1] buffer = None
+    if isinstance(filepath_or_buffer, BytesIO):
+        buffer = filepath_or_buffer.getbuffer()
+    elif isinstance(filepath_or_buffer, StringIO):
+        buffer = filepath_or_buffer.read().encode()
+    elif isinstance(filepath_or_buffer, bytes):
+        buffer = filepath_or_buffer
```

Comment:
Should this be put into a utility function and shared across the JSON reader and any other readers?

Reply:
Yup, good idea